src/adapter/src/coord.rs

```diff
@@ -775,8 +818,9 @@ impl<S: Append + 'static> Coordinator<S> {
         }
 
         // Having installed all entries, creating all constraints, we can now relax read policies.
-        self.initialize_read_policies(&policies_to_set, DEFAULT_LOGICAL_COMPACTION_WINDOW_MS)
-            .await;
+        for (ts, policies) in policies_to_set {
+            self.initialize_read_policies(&policies, Some(ts)).await;
+        }
```

Comment:
Can you add a comment about what we discussed for this that says we want to eliminate the loop in the future?